segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/PravegaRequestProcessor.java

```diff
@@ -552,6 +571,144 @@ public void updateSegmentPolicy(UpdateSegmentPolicy updateSegmentPolicy) {
+            return;
+        }
+
+        log.info(deleteTableSegment.getRequestId(), "Deleting table segment {}.", deleteTableSegment);
+        tableStore.deleteSegment(segment, deleteTableSegment.isMustBeEmpty(), TIMEOUT)
+                  .thenRun(() -> connection.send(new SegmentDeleted(deleteTableSegment.getRequestId(), segment)))
+                  .exceptionally(e -> handleException(deleteTableSegment.getRequestId(), segment, operation, e));
+    }
+
+    @Override
+    public void mergeTableSegments(final MergeTableSegments mergeTableSegments) {
+        final String operation = "mergeTableSegments";
+
+        if (!verifyToken(mergeTableSegments.getSource(), mergeTableSegments.getRequestId(), mergeTableSegments.getDelegationToken(), operation)) {
+            return;
+        }
+
+        log.info(mergeTableSegments.getRequestId(), "Merging table segments {}.", mergeTableSegments);
+        tableStore.merge(mergeTableSegments.getTarget(), mergeTableSegments.getSource(), TIMEOUT)
+                  .thenRun(() -> connection.send(new WireCommands.SegmentsMerged(mergeTableSegments.getRequestId(),
+                                                                                 mergeTableSegments.getTarget(),
+                                                                                 mergeTableSegments.getSource())))
+                  .exceptionally(e -> handleException(mergeTableSegments.getRequestId(), mergeTableSegments.getSource(), operation, e));
+    }
+
+    @Override
+    public void sealTableSegment(final WireCommands.SealTableSegment sealTableSegment) {
+        String segment = sealTableSegment.getSegment();
+        final String operation = "sealTableSegment";
+
+        if (!verifyToken(segment, sealTableSegment.getRequestId(), sealTableSegment.getDelegationToken(), operation)) {
+            return;
+        }
+
+        log.info(sealTableSegment.getRequestId(), "Sealing table segment {}.", sealTableSegment);
+        tableStore.seal(segment, TIMEOUT)
+                  .thenRun(() -> connection.send(new SegmentSealed(sealTableSegment.getRequestId(), segment)))
+                  .exceptionally(e -> handleException(sealTableSegment.getRequestId(), segment, operation, e));
+    }
+
+    @Override
+    public void updateTableEntries(final WireCommands.UpdateTableEntries updateTableEntries) {
+        String segment = updateTableEntries.getSegment();
+        final String operation = "updateTableEntries";
+
+        if (!verifyToken(segment, updateTableEntries.getRequestId(), updateTableEntries.getDelegationToken(), operation)) {
+            return;
+        }
+
+        log.info(updateTableEntries.getRequestId(), "Updating table segment {}.", updateTableEntries);
+        List<TableEntry> entries = updateTableEntries.getTableEntries().getEntries().stream()
+                                                     .map(e -> versioned(new ByteArraySegment(e.getKey().getData()),
```

Comment:
Not all of these are `versioned`. We need to make the following transformations:
1. If `Client.TableKey.getVersion() == null` -> use `TableEntry.unversioned`
2. If `Client.TableKey.getVersion() == io.pravega.client.tables.impl.TableKey.NOT_EXISTS` -> use `TableEntry.notExists`
3. Else use `TableKey.versioned`

It just occurred to me that ` io.pravega.client.tables.impl.TableKey.NOT_EXISTS` is null, which is incorrect (and has a TODO associated with it). We should set that to a different value. Let's say it has a version matching `Long.MIN_VALUE` so it matches the server-side.

Reply:
My understanding was that the Table entry can have 3 possible versions.
a.  `TableKey.NO_VERSION `  (Long.MIN_VALUE)
b. `TableKey.NOT_EXISTS` ( -1L)
c. actual version. Any value >= 0L.
These are the 3 values which will always be sent inside a WireCommand and null is never passed as part of the WireCommand. Hence the `TableEntry.versioned(..)` api is used.

Reply:
Ah I see. Ok, then we should handle this when implementing the client-side contracts.